src/common/droid-util.c

```diff
@@ -1062,6 +1072,8 @@ static pa_droid_hw_module *droid_hw_module_open(pa_core *core, const pa_droid_co
         pa_shared_set(core, DROID_HW_HANDLE_V1, hw);
         pa_shared_set(core, DROID_SET_PARAMETERS_V1, droid_set_parameters_v1_cb);
         pa_shared_set(core, DROID_GET_PARAMETERS_V1, droid_get_parameters_v1_cb);
+
+        hw_primary = hw;
```

Comment:
any chance that those pa_shared_set() calls could now or in the future span somewhere where hw_primary is used - i.e. should the assignment be done 1st?

Reply:
There's always a chance :) Changed order.